File: app/src/main/java/doubleslash05/mini/team11/ui/tutorial/TutorialActivity.java
package doubleslash05.mini.team11.ui.tutorial;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;
import android.os.Handler;

import com.google.android.material.tabs.TabLayout;

import doubleslash05.mini.team11.R;



public class TutorialActivity extends AppCompatActivity {



    FragmentPagerAdapter adapterViewPager;


    public static class TutorialPageAdapter extends FragmentPagerAdapter {
        private static int PAGE_NUN = 5;



        @Override
        public int getItemPosition(@NonNull Object object) {
            return super.getItemPosition(object);
        }




        public TutorialPageAdapter(FragmentManager fragmentManager) {



            super(fragmentManager);


        }

        @Override
        public int getCount() {
            return PAGE_NUN;
        }

        @NonNull
        @Override
        public Fragment getItem(int position) {
            switch (position) {
                case 0:
                    TutorialFragment1 tutorialFragment1 = new TutorialFragment1();
                    return tutorialFragment1;
                case 1:
                    TutorialFragment2 tutorialFragment2 = new TutorialFragment2();
                    return tutorialFragment2;
                case 2:
                    TutorialFragment3 tutorialFragment3 = new TutorialFragment3();
                    return tutorialFragment3;
                case 3:
                    TutorialFragment4 tutorialFragment4 = new TutorialFragment4();
                    return tutorialFragment4;
                case 4:
                    TutorialFragment5 tutorialFragment5 = new TutorialFragment5();
                    return tutorialFragment5;
                default:
                    return null;

            }
        }

        @Nullable
        @Override
        public CharSequence getPageTitle(int position) {
            return null;

        }
    }





    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_tutorial);
        final ViewPager vpPager = (ViewPager)findViewById(R.id.vpPager);
        TabLayout customTab = (TabLayout)findViewById(R.id.customTab);

        customTab.setupWithViewPager(vpPager, true);

        adapterViewPager = new TutorialPageAdapter(getSupportFragmentManager());
        vpPager.setAdapter(adapterViewPager);
        vpPager.setCurrentItem(0);


        new Handler().postDelayed(new Runnable()
        {
            @Override
            public void run()
            {

                vpPager.setCurrentItem(1);
            }
        },  5000);





    }





}

File: app/src/main/java/doubleslash05/mini/team11/ui/tutorial/TutorialActivity.kt
package doubleslash05.mini.team11.ui.tutorial

import android.Manifest
import android.annotation.SuppressLint
import android.content.pm.PackageManager
import android.os.Bundle
import android.os.Handler
import android.util.Log
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.icaksama.rapidsphinx.RapidPreparationListener
import com.icaksama.rapidsphinx.RapidSphinx
import com.icaksama.rapidsphinx.RapidSphinxListener
import doubleslash05.mini.team11.R
import doubleslash05.mini.team11.ui.base.BaseActivity
import doubleslash05.mini.team11.ui.base.BaseFragment
import doubleslash05.mini.team11.ui.main.HomeActivity
import edu.cmu.pocketsphinx.Config
import kotlinx.android.synthetic.main.activity_tutorial.*
import org.jetbrains.anko.startActivity

class TutorialActivity : BaseActivity(), RapidSphinxListener {
    companion object {
        private const val pauseKeyword = "darcy"
        private const val stopKeyword = "stop"
        private const val stopKeyword2 = "region"
        private const val nextKeyword = "daum"
        private const val start = "시작하기"
    }

    private val rapidSphinx: RapidSphinx by lazy { RapidSphinx(this) }
    private val adapterViewPager: FragmentPagerAdapter = TutorialPageAdapter(supportFragmentManager)
    private val startFragment = Tutorial4Fragment()
    private val updateHandler = Handler()
    private val runnable = Runnable {
        updateDisplay()
    }

    override fun rapidSphinxDidStop(reason: String, code: Int) {
        rapidSphinx.startRapidSphinx(10000)
    }

    override fun rapidSphinxFinalResult(
        result: String,
        hypArr: List<String>,
        scores: List<Double>
    ) {

    }

    @SuppressLint("ResourceType")
    override fun rapidSphinxPartialResult(partialResult: String) {

        when (partialResult) {
            stopKeyword2 -> {
                viewpager_tutorial.currentItem = 1
                btnSkip.setText("Skip")
            }
            nextKeyword -> {
                viewpager_tutorial.currentItem = 2
            }
            pauseKeyword -> {
                supportFragmentManager.beginTransaction()
                    .replace(R.id.fragment_tutorial_main, startFragment)
                    .addToBackStack(null).commit()
                btnSkip.setText(start)
                rapidSphinx.stop()
            }

        }


    }

    override fun rapidSphinxUnsupportedWords(words: List<String>) {

    }

    override fun rapidSphinxDidSpeechDetected() {

    }


    private var goToHome = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_tutorial)
        btnSkip.isEnabled = false
        updateHandler.postDelayed(runnable, 5000)



        btnSkip.setOnClickListener {
            if (goToHome) {
                startActivity<HomeActivity>()
            } else if (viewpager_tutorial.currentItem == 0) {
                btnSkip.setText("Skip")
                viewpager_tutorial.currentItem += 1
                btnSkip.isEnabled = false
                updateHandler.postDelayed(runnable, 5000)
                btnSkip.setBackgroundResource(R.drawable.btn_top_rounded)


            } else if (viewpager_tutorial.currentItem == 1) {
                btnSkip.setText("Skip")
                viewpager_tutorial.currentItem += 1
                btnSkip.isEnabled = false
                updateHandler.postDelayed(runnable, 5000)
                btnSkip.setBackgroundResource(R.drawable.btn_top_rounded)


            } else if (viewpager_tutorial.currentItem == 2) {
                btnSkip.isSelected = true
                supportFragmentManager.beginTransaction()
                    .replace(R.id.fragment_tutorial_main, startFragment)
                    .addToBackStack(null).commit()
                btnSkip.setText(start)
                btnSkip.isEnabled = true
                btnSkip.isClickable = true
                rapidSphinx.stop()
                goToHome = true
            }


        }
        customTab.setupWithViewPager(viewpager_tutorial, true)
        viewpager_tutorial.adapter = adapterViewPager
        Log.d("cycle", "onCreate")
        rapidSphinx.addListener(this)



        if (isPermissionsGranted)
            rapidSphinx.prepareRapidSphinx(object : RapidPreparationListener {
                override fun rapidPreExecute(config: Config) {
                    rapidSphinx.isRawLogAvailable = true
                    config.setString("-logfn", "/dev/null")
                    config.setBoolean("-verbose", true)
                }

                override fun rapidPostExecute(isSuccess: Boolean) {
                }
            })

    }

    override fun onResume() {
        super.onResume()
        Log.d("cycle", "onResume")
        val oovwords = arrayOf("darcy", "daum", "region")
        rapidSphinx.updateVocabulary(
            stopKeyword,
            oovwords
        ) { rapidSphinx.startRapidSphinx(10000) }

    }

    private val isPermissionsGranted: Boolean
        private get() = if (ContextCompat.checkSelfPermission(
                this,
                Manifest.permission.RECORD_AUDIO
            ) != PackageManager.PERMISSION_GRANTED
        ) {
            ActivityCompat.requestPermissions(
                this, arrayOf(Manifest.permission.RECORD_AUDIO),
                1
            )
            false
        } else {
            true
        }

    private fun updateDisplay() {
        btnSkip.isEnabled = true
        btnSkip.setBackgroundResource(R.drawable.btn_top_rounded_activated)

    }

    class TutorialPageAdapter(fragmentManager: FragmentManager?) : FragmentPagerAdapter(
        fragmentManager!!
    ) {

        private val PAGE_NUMBER = listOf<BaseFragment>(
            Tutorial1Fragment(),
            Tutorial2Fragment(),
            Tutorial3Fragment()
        )


        override fun getItemPosition(`object`: Any): Int {
            return super.getItemPosition(`object`)
        }

        override fun getCount(): Int {
            return PAGE_NUMBER.size
        }

        override fun getItem(position: Int): Fragment {
            return PAGE_NUMBER[position]
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return null
        }


    }


}